ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -1260,6 +1290,89 @@ private void testHardDeletePauseResume(boolean reloadIndex) throws InterruptedEx
+    // fill current log segment to its capacity
+    fillLastLogSegmentToCapacity(1, false);
+
+    // test every log segment except the last one is returned
+    fillLastLogSegmentToCapacity(maxEntriesInJournal, true);
+    assertEquals("LogSegments mismatch ", getExpectedLogSegmentNames(1), state.index.getLogSegmentsNotInJournal());
+
+    // create maxLogSegmentsToBeIgnored
+    int logSegmentCount = maxLogSegmentsToBeIgnored;
+    if (maxEntriesInJournal % maxLogSegmentsToBeIgnored > 0) {
+      logSegmentCount--;
+    }
+    for (int i = 0; i < logSegmentCount; i++) {
+      fillLastLogSegmentToCapacity(maxEntriesInJournal / maxLogSegmentsToBeIgnored, true);
+    }
+    if (maxEntriesInJournal % maxLogSegmentsToBeIgnored > 0) {
+      fillLastLogSegmentToCapacity(
+          maxEntriesInJournal - (logSegmentCount * (maxEntriesInJournal / maxLogSegmentsToBeIgnored)), true);
+    }
+    // every log segment except the last "maxLogSegmentsToBeIgnored" number of log segments should be returned
+    assertEquals("LogSegments mismatch ", getExpectedLogSegmentNames(maxLogSegmentsToBeIgnored),
+        state.index.getLogSegmentsNotInJournal());
+
+    if (reloadIndex) {
+      state.reloadIndex(true, false);
+      // every log segment except the last one should be returned
+      assertEquals("LogSegments mismatch ", getExpectedLogSegmentNames(1), state.index.getLogSegmentsNotInJournal());
+    }
+  }
+
+  /**
+   * Fetch expected log segment names ignoring the last {@code segmentsToIgnoreFromLast} from the list
+   * of actual log segments in the log
+   * @param segmentsToIgnoreFromLast number of log segments to be ignored from the last
+   * @return a {@link List} of log segments
+   */
+  private List<String> getExpectedLogSegmentNames(int segmentsToIgnoreFromLast) {
+    List<String> expectedLogSegments = new ArrayList<>();
+    LogSegment logSegment = state.log.getFirstSegment();
+    while (logSegment != null) {
+      expectedLogSegments.add(logSegment.getName());
+      logSegment = state.log.getNextSegment(logSegment);
+    }
+    return expectedLogSegments.subList(0, expectedLogSegments.size() - segmentsToIgnoreFromLast);
+  }
+
+  /**
+   * Fill the current log segment to it capacity
+   * @param numberOfEntries number of entries to be added to fill the log segment to its capacity
+   * @param newLogSegment {@code true} if this is a new log segment. {@code false} otherwise
+   * @throws InterruptedException
+   * @throws StoreException
+   * @throws IOException
+   */
+  private void fillLastLogSegmentToCapacity(int numberOfEntries, boolean newLogSegment)
+      throws InterruptedException, StoreException, IOException {
+    if (newLogSegment) {
+      // to ensure a new log segment is created. If not, find the remaining size below will fail
+      state.addPutEntries(1, PUT_RECORD_SIZE, Utils.Infinite_Time);
+      numberOfEntries--;
+    }
+    // 1 PUT entry that spans the rest of the data in the last segment
```

Comment:
this comment seems out of place. More than 1 is inserted based on the arguments right?